### DISCUSSION TITLE
Question on package publishing

### DISCUSSION THREAD

**Discussion by bbohling** [verdaccio/verdaccio#2980] [Q&A]
We are using the standard verdaccio container (w/version 5.5.2) with the LDAP plugin. While we can `npm publish` just fine, what has us baffled is why any authenticated user can publish to any existing package. So if dev1 publishing widget1, dev2 can also publish to widget1. Surely there is a way to lock this down, but the Best Practices page doesn't seem to cover that. I would expect dev2 to not be able to publish widget1 unless dev1 explicitly sets dev2 up with `npm owner add`.  

I do see under [Compatibility](https://github.com/verdaccio/verdaccio#compatibility) that "Transferring ownership" is not supported, but does that mean Verdaccio truly is only for teams not an enterprise? I suppose unless you managed "ownership" by scope, but that's not realistic for a large company.

**Comment by edclement** [comment]
Are you able to provide an example of what your [package access](https://verdaccio.org/docs/packages) permissions look like? I'm guessing you are using the [verdaccio-ldap](https://www.npmjs.com/package/verdaccio-ldap) plugin? The official Verdaccio docs do mention:
> The security constraints remain on the shoulders of the plugin being used, by default verdaccio uses the [htpasswd plugin](https://github.com/verdaccio/verdaccio-htpasswd). If you use a different plugin the behaviour might be different.

Looking at the README for the `verdaccio-ldap` plugin, there is no real mention of how/if it respects the same security constraints as the `htpasswd` plugin. I would recommend digging into the code a little for the `verdaccio-ldap` plugin to see what it is doing. I took a quick look at the code and don't see the [`allow_*`](https://verdaccio.org/docs/plugin-auth/#allow_access-allow_publish-or-allow_unpublish-callback) callbacks implemented so that is likely where the problem resides (best guess). I'm guessing Verdaccio uses some default behavior since those callbacks are not implemented in the plugin.
